documentation/specs/proposed/wasmtask.wit

```diff
@@ -1,62 +1,88 @@
-﻿package msbuildtask:v0
+﻿package MSBuildWasm:v0
 
-record str-kv-pair {
-    key: string
-    value: string
+// WIT doesn't have maps/dictionaries
+record StringKVPair {
+    Key: string
+    Value: string
 }
+type StringMap = list<StringKVPair>
 
-type str-dictionary = list<str-kv-pair>
+// mirroring TaskItem in MSBuild - it's a description of a file or directory
+record TaskItem { 
+    // Path inside the sandbox
+    WasmPath: string;
+    Metadata: StringMap
+}
+// enum for types of properties in Tasks
+enum PropertyType {
+    BoolType,
+    StringType,
+    TaskItemType,
+    BoolArrayType,
+    StringArrayType,
+    TaskItemArrayType
+}
+
+// variant contains one of the possible types of properties
+variant PropertyValue {
+    Bool(bool);
+    String(string);
+    TaskItem(TaskItem);
+    BoolArray(BoolList);
+    StringArray(StringList);
+    TaskItemArray(TaskItemList);
+}
 
-record TaskItem {
-    ItemSpec: string
-    Metadata: str-dictionary
+type TaskItemList = list<TaskItem>
+type BoolList = list<bool>
+type StringList = list<string>
+
+// mirrors MSBuild's TaskPropertyInfo
+record TaskPropertyInfo {
+    Name: string;
+    Type: PropertyType;
+    Output: bool;
+    Required: bool;
 }
 
-type TaskItems = list<TaskItem>
-type Bools = list<bool>
-type Strings = list<string>
+// information about a task passed from the Task to MSBuild
+record TaskInfo {
+    Name: string; 
+    Parameters: list<TaskPropertyInfo>; 
+    Metadata: StringMap; // e.g. requirements for the host environment
+}
 
-interface host-data {
-    GetHostData: func() -> str-dictionary
+// information about the host environment passed from MSBuild to the Task
+record HostInfo {
+    PreopenedDirectories: TaskItemList;
 }
 
+// input to Execute 
+// (in MSBuild normally the task already magically has set its properties to the values from the project file, but here we need to pass them explicitly)
```

Comment:
I like this change, taking it from "create instance of object, set properties, call method" to a single function call is very nice.